src/compression/mod.rs

```diff
@@ -166,8 +166,8 @@ impl Compression {
     pub fn compress_image_section(self, header: &Header, mut uncompressed: ByteVec, pixel_section: IntegerBounds) -> Result<ByteVec> {
         let max_tile_size = header.max_block_pixel_size();
 
-        assert!(pixel_section.validate(Some(max_tile_size)).is_ok(), "decompress tile coordinate bug");
-        if header.deep { assert!(self.supports_deep_data()) }
+        debug_assert!(pixel_section.validate(Some(max_tile_size)).is_ok(), "decompress tile coordinate bug");
```

Comment:
Ok to not have in release mode, because low level api and because not exr specific